project/src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -69,10 +70,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
           true /* filter if no website */,
           true /* filter branches of same place */
       );
-    } catch (Exception e) {
+    } catch (FetcherException e) {
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
           "Fetching from Google Places API encountered a problem");
       return;
+    } catch (IllegalArgumentException e) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
```

Comment:
This is most definitely not an `SC_INTERNAL_SERVER_ERROR`, but rather an `SC_BAD_REQUEST`. 

When returning an HTTP response, it's important to distinguish between server and client error. 
A server error is your way of telling the client: "I had a problem, my bad!". The client can go ahead and retry in a short while, and potentially get a response. These are all represented as 500 errors. The sub errors (5XX) represent more nuanced view into what happened. For instance, 503 says that server is unavailable (which leads to network/setup/etc. problems), while 501 (NOT_IMPLEMENTED) tells the client that this functionality does not exist in the server (so it shouldn't retry the reuqest!)

A client error is when the client did something wrong. Like - provide wrong arguments. Retrying the request won't do any good here! They should look into what they did wrong and fix the problem.

Reply:
That's a great point, thanks for clearing that. Changed.